query/src/main/java/org/infinispan/query/impl/EntityLoader.java

```diff
@@ -17,54 +12,20 @@
  * @author Marko Luksa
  * @since 5.0
  */
-public final class EntityLoader<E> implements PojoSelectionEntityLoader<EntityLoaded<E>> {
+public final class EntityLoader<E> extends BaseEntityLoader<E, Object> {
 
-   private final AdvancedCache<?, E> cache;
-   private final LocalQueryStatistics queryStatistics;
 
-   EntityLoader(AdvancedCache<?, E> cache, LocalQueryStatistics queryStatistics) {
-      this.cache = cache;
-      this.queryStatistics = queryStatistics;
+   EntityLoader(AdvancedCache<Object, Object> cache, LocalQueryStatistics queryStatistics) {
+      super(cache, queryStatistics);
    }
 
    @Override
-   public List<EntityLoaded<E>> loadBlocking(List<?> identifiers, Deadline deadline) {
-      if (identifiers.isEmpty()) return Collections.emptyList();
-
-      int entitiesSize = identifiers.size();
-      LinkedHashSet<Object> keys = new LinkedHashSet<>(entitiesSize);
-      for (Object identifier : identifiers) {
-         keys.add(cache.getKeyDataConversion().fromStorage(identifier));
-      }
-
-      long start = queryStatistics.isEnabled() ? System.nanoTime() : 0;
-
-      // getAll instead of multiple gets to get all the results in the same call
-      Map<?, E> values = cache.getAll(keys);
-
-      if (queryStatistics.isEnabled()) queryStatistics.entityLoaded(System.nanoTime() - start);
-
-      ArrayList<EntityLoaded<E>> result = new ArrayList<>(entitiesSize);
-      for (Object key : keys) {
-         // if the entity was present at indexing time and
-         // it is not present anymore now at searching time,
-         // we will add a null here
-         result.add(new EntityLoaded<>(values.get(key), null));
-      }
-
-      return result;
-   }
-
-   @Override
-   public boolean equals(Object o) {
-      if (this == o) return true;
-      if (o == null || getClass() != o.getClass()) return false;
-      EntityLoader<?> that = (EntityLoader<?>) o;
-      return Objects.equals(cache, that.cache) && Objects.equals(queryStatistics, that.queryStatistics);
+   Map<Object, Object> loadEntries(List<?> keys) {
+      return cache.withStorageMediaType().getAll(Set.copyOf(keys));
```

Comment:
Do we care about the order here as well? The getAll command technically supports the order of the Set passed in as well.